cirq-core/cirq/contrib/acquaintance/executor.py

```diff
@@ -100,6 +117,67 @@ def optimization_at(
         )
 
 
+class StrategyExecutorTransformer:
+    """Executes an acquaintance strategy."""
+
+    def __init__(self, execution_strategy: ExecutionStrategy) -> None:
+        """Initializes transformer.
+
+        Raises:
+            ValueError: if execution_strategy is None.
+        """
+
+        if execution_strategy is None:
+            raise ValueError('execution_strategy cannot be None')
+        self.execution_strategy = execution_strategy
+        self.mapping = execution_strategy.initial_mapping.copy()
+
+    def __call__(
+        self, circuit: circuits.Circuit, context: Optional['cirq.TransformerContext'] = None
+    ) -> circuits.Circuit:
+        """Executes an acquaintance strategy using primitive map function and
+          mutates initial mapping.
+
+        Args:
+            circuit: 'circuits.Circuit' input circuit to transform.
+            context: `cirq.TransformerContext` storing common configurable
+              options for transformers.
+
+        Returns:
+            A copy of the modified circuit after executing an acquaintance
+              strategy on all instances of AcquaintanceOpportunityGate
+        """
+
+        expose_acquaintance_gates(circuit)
```

Comment:
For this method to be able to work on `cirq.AbstractCircuit`, you'd have to update the `expose_acquaintance_gates(circuit)` because this call currently expects a `cirq.Circuit` to work on. 

This can be done easily by:

```suggestion
        circuit = transformers.expand_composite(circuit, no_decomp= expose_acquaintance_gates.no_decomp)
```

Reply:
Okay!